### ISSUE TITLE
Define primary hosted zones in code, with CI/CD

### ISSUE THREAD

**Issue by ohaiwalt** [OperationCode/operationcode_infra#67]
Similar to that described in #2, we recently had an event where a few people wanted a DNS record and didn't know where to go.

It should be possible to manage DNS records for operationcode.org from within this repository.

To do so:

- [x] Create an S3 bucket for Terraform remote state storage.

- [x] Create AWS user credentials to provide to CircleCI, along with an IAM policy providing access to the bucket for remote state storage.

- [x] Get the DNSimple API key and provide it to CircleCI.

- [x] Use Terraform to define records as they currently exist in DNSimple. DNSimple credentials are in 1Password.

- [ ] Run `terraform import` on all existing resources once the configuration is complete, and run `terraform plan` to validate. We should see no changes.

- [x] Create a record by adding a new Terraform resource definition, and on merge to master it should apply.

Bonus points if `terraform plan` is run against the PR prior to merging. Consider using `terragrunt`.

**Comment by ohaiwalt** [comment]
interestingly, found that DNSimple apply will fail after a record is configured in Terraform if it already exists.

**Comment by ohaiwalt** [comment]
I'm not going to exhaustively describe our DNS resources at this moment, I'll open another issue to transpose that info.
